### PR TITLE
feat(table-of-contents): added horizontal variant

### PR DESCRIPTION
**Pull request by RaphaelAmadeu**
### Related Ticket(s)

#5072

### Description

I've added the Horizontal variant to our `TableOfContents`, by using the newly created `type` property.

image

### Changelog

**New**

- Horizontal variant

### CODE REVIEW

File: packages/web-components/src/components/table-of-contents/table-of-contents.ts [hunk 1]
```diff
@@ -275,9 +282,20 @@ class DDSTableOfContents extends StableSelectorMixin(LitElement) {
...
     } = this;
+
+    const containerClasses = classMap({
+      [`${ddsPrefix}-ce--table-of-contents__container`]: this.type === TOC_TYPES.DEFAULT,
+      [`${ddsPrefix}-ce--table-of-contents-horizontal__container`]: this.type === TOC_TYPES.HORIZONTAL,
+    });
+
+    const navigationClasses = classMap({
+      [`${prefix}--tableofcontents__sidebar`]: this.type === TOC_TYPES.DEFAULT,
+      [`${prefix}--tableofcontents__horizontal`]: this.type === TOC_TYPES.HORIZONTAL,
```
Comment (asudoh) [inline packages/web-components/src/components/table-of-contents/table-of-contents.ts:293]: `__navbar` or `__topbar`? Naïvely I thought of suggesting `--horizontal` as "horizontal" means a style variant.

File: packages/web-components/src/components/table-of-contents/table-of-contents.ts [hunk 2]
```diff
@@ -30,6 +31,12 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
  */
 @customElement(`${ddsPrefix}-table-of-contents`)
 class DDSTableOfContents extends StableSelectorMixin(LitElement) {
+  /**
+   * Defines TOC type, `` for default, `horizontal` for horizontal variant.
+   */
+  @property({ reflect: true })
+  type = TOC_TYPES.DEFAULT;
```
Comment (asudoh) [inline packages/web-components/src/components/table-of-contents/table-of-contents.ts:38]: We need more descriptive name, an quick idea is `toc-layout`.

**Review by asudoh — COMMENTED** [review]
Thank you for the work @RaphaelAmadeu!

**Comment by jeffchew** [comment]
This looks great @RaphaelAmadeu !

There's a minor thing, the documentation formatting looks a bit off on the description for `type`:

Screen Shot 2021-03-11 at 9 53 45 AM

**Review by jeffchew — APPROVED** [review]
Looks great, thank you @RaphaelAmadeu !

**Comment by oliviaflory** [comment]
Hey @RaphaelAmadeu this is looking really good! I have a few comments:

1. In the selected state, text color should change to $text-01, looks like it's still $text-02.
Screen Shot 2021-03-11 at 5 28 24 PM


Screen Shot 2021-03-11 at 5 05 12 PM


2. Looks like the selected state blue bar is grayed out on hover, can we remove that?
Screen Shot 2021-03-11 at 5 03 11 PM

Screen Shot 2021-03-11 at 5 24 37 PM



3. The heading is hidden behind the Horizontal TOC when clicking through the options, could we clear it by 80px?. Seeing the heading is really important for the user experience to associate the TOC label with the section they're jumping to. Looks like this is already working on the mobile breakpoints 

https://user-images.githubusercontent.com/45692622/110863221-5c462580-828e-11eb-92e6-969d7084d536.mp4



4. Web components Default TOC is missing the scroll animation feature that is present in [React](https://ibmdotcom-react.mybluemix.net/?path=/story/components-table-of-contents--manually-define-menu-items).



5. Looks like the text is stacking to two lines between 1055–1160px, we had the overflow feature from the masthead in the specs. Realizing this might be a bit tight to get into the release and fully test, so LMK if it's too much, maybe we could have the Select kick in early if the content runs out of room?
Screen Shot 2021-03-11 at 5 06 34 PM

Screen Shot 2021-03-11 at 5 27 06 PM


6. Is there any way to see the Horizontal TOC and the Masthead on the same page to QA the interaction?

**Pull request merged**